tests/test_command.py

```diff
@@ -798,3 +798,15 @@ def test_help_text(self):
                             # not too long
                             assert len(help_text) < 80
         assert not commands, "Commands without help text: %s" % commands
+    
+    @mock.patch("alembic.command.os.listdir", return_value = ["file1", "file2"])
+    @mock.patch("alembic.command.os.access", return_value = True)
+    def test_init_file_exists_and_is_not_empty(self, mocked_listdir, mocked_access):
+        directory = "alembic"
+        assert_raises_message(
+            util.CommandError,
+            "Directory %s already exists and is not empty" % directory,
+            command.init,
+            self.cfg,
+            directory = directory
+        )
```

Comment:
**mike bayer** (zzzeek) wrote:

OK can you add two more tests, which you just have to change what you're patching:

test_init_file_exists_and_is_empty
test_init_file_doesnt_exist

also, stylistically I'd prefer if the mock.patch is on the inside, like:

    def test_thing(self):
        with mock.patch(
            "alembic.command.os.listdir", return_value=["file1", "file2"]
        ), mock.patch("alembic.command.os.access", return_value=True):
            pass